py-jproperties: add v1.0.1, v2.0.0, v2.1.0, v2.1.1

teaguesterling opened the pull request:
Add a few new and older versions of the `py-jproperties` package. Older versions are needed as dependencies for upcoming packages.

teaguesterling committed:
py-uvloop: add v3.8.14, v3.9.15, v3.10.3 and dependencies

teaguesterling committed:
rollback

teaguesterling committed:
py-jproperties: add v1.0.1, v2.0.0, v2.1.0, v2.1.1

teaguesterling committed:
cleanup

teaguesterling commented:
@spackbot fix style

teaguesterling committed:
Update package.py

teaguesterling committed:
Update package.py

tldahlgren commented on var/spack/repos/builtin/packages/py-jproperties/package.py:
```diff
@@ -0,0 +1,27 @@
+    version("1.0.1", sha256="327e14082653a4f2212ff81a96fbf141382f727f421e8afc933bf56ff7c010f4")
+
+    depends_on("python@2.7,3:", type=("build", "run"))
+    depends_on("py-six@1.13:", type=("build", "run"))
```
Looks like `1.0.1` doesn't have this dependency.

While `2.0.0`'s `setup.py` has:
```
    install_requires=["six ~= 1.10"],
    setup_requires=["pytest-runner ~= 2.0"],
```
but only `six` listed in the egg's `requires.txt`.  The `six` requirement should correspond to `py-six@1.10:1`.

`2.1.0`'s `setup.py` has:
```
    install_requires=["six ~= 1.12"],
    setup_requires=["pytest-runner ~= 2.0"],
```
so `six` should correspond to `py-six@1.12:1`.

Finally, `2.1.1`'s `setup.py` has the build constraints:
```
    install_requires=["six ~= 1.13"],
    setup_requires=[
        "setuptools_scm ~= 3.3"
    ],
```
which corresponds to `py-six@1.13:1` plus the additional dependency of `py-setuptools-scm@3.3:3`. Though the egg's `requires.txt` only shows the six dependency.

teaguesterling replied:
Thank you! Will add these in. I really appreciate the time to capture this. It helps me think through what to look for in future PRs.

tldahlgren reviewed (changes requested):
Confirmed the homepage and version sha256s but I think the dependencies for the versions need a bit more work.

teaguesterling committed:
Update package.py

Incorporating changes from review: https://github.com/spack/spack/pull/44519#discussion_r1625170596

teaguesterling committed:
Update package.py

tldahlgren reviewed (approved):
Thanks for the updates.
